tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,237 @@
 """Tests for ZHA config flow."""
-from unittest import mock
 
+import os
+
+import pytest
+import serial.tools.list_ports
+import zigpy.config
+
+from homeassistant import setup
 from homeassistant.components.zha import config_flow
-from homeassistant.components.zha.core.const import CONTROLLER, DOMAIN, ZHA_GW_RADIO
-import homeassistant.components.zha.core.registries
+from homeassistant.components.zha.core.const import CONF_RADIO_TYPE, CONTROLLER, DOMAIN
+from homeassistant.components.zha.core.registries import RADIO_TYPES
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_SOURCE
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
 
-import tests.async_mock
+from tests.async_mock import AsyncMock, MagicMock, patch, sentinel
 from tests.common import MockConfigEntry
 
 
-async def test_user_flow(hass):
-    """Test that config flow works."""
-    flow = config_flow.ZhaFlowHandler()
-    flow.hass = hass
+def com_port():
+    """Mock of a serial port."""
+    port = serial.tools.list_ports_common.ListPortInfo()
+    port.serial_number = "1234"
+    port.manufacturer = "Virtual serial port"
+    port.device = "/dev/ttyUSB1234"
+    port.description = "Some serial port"
 
-    with tests.async_mock.patch(
-        "homeassistant.components.zha.config_flow.check_zigpy_connection",
-        return_value=False,
-    ):
-        result = await flow.async_step_user(
-            user_input={"usb_path": "/dev/ttyUSB1", "radio_type": "ezsp"}
-        )
+    return port
 
-    assert result["errors"] == {"base": "cannot_connect"}
 
-    with tests.async_mock.patch(
-        "homeassistant.components.zha.config_flow.check_zigpy_connection",
-        return_value=True,
-    ):
-        result = await flow.async_step_user(
-            user_input={"usb_path": "/dev/ttyUSB1", "radio_type": "ezsp"}
-        )
+@patch("serial.tools.list_ports.comports", MagicMock(return_value=[com_port()]))
+@patch(
+    "homeassistant.components.zha.config_flow.detect_radios",
+    return_value={CONF_RADIO_TYPE: "test_radio"},
+)
+async def test_user_flow(detect_mock, hass):
+    """Test user flow -- radio detected."""
 
-    assert result["type"] == "create_entry"
-    assert result["title"] == "/dev/ttyUSB1"
-    assert result["data"] == {"usb_path": "/dev/ttyUSB1", "radio_type": "ezsp"}
+    port = com_port()
+    port_select = f"{port}, s/n: {port.serial_number} - {port.manufacturer}"
 
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data={zigpy.config.CONF_DEVICE_PATH: port_select},
+    )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"].startswith(port.description)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)